elearning-frontend/src/components/admin/EditWords.js

```diff
@@ -0,0 +1,331 @@
+                    return {
+                        choice: (currentIndex === selectedIndex) ? input : value.choice,
+                        is_correct: value.is_correct,
+                        label: value.label
+                    }
+                }).reduce(labelArray, {})
+        })
+    }
+
+    // Set checkbox states
+    const setCorrectAnswer = (selectedIndex, type, input) => {
+        setWord({
+            ...word,
+            choices:
+                Object.entries(word.choices).map(([key, value], index) => {
+                    var currentIndex = index + 1
+                    return {
+                        choice: value.choice,
+                        is_correct: (currentIndex === selectedIndex) ? !value.is_correct : false,
+                        label: value.label
+                    }
+                }).reduce(labelArray, {})
+        })
+    }
+
+    var view_element = ""
+    var dropdown_options = ""
+
+    if (changeWordData) {
+        view_element = <tr><td colSpan={5} className='text-center'> LOADING DATA</td></tr>
+    }
+    else {
+        view_element =
+        wordList.map((word) => {
+            return <tr key={ word.id }>
+                <td className='text-center'>{ word.category.title }</td>
+                <td className='text-center'>{ word.word }</td>
+                <td className='text-center'>
+                    {
+                        Object.entries(word.choices).map(([key, value], index) => {
+                            return <p key={ key }
+                                        className={`text-center ${index % 2 == 0 ? 'bg-slate-200' : ''}`}>
+                                        { value.choice }
+                                    </p>
+                        })
+                    }
+                </td>
+                <td className='text-center'>{ word.correct_answer }</td>
+                <td className='flex'>
+                    <button onClick={(e) => {
+                        e.preventDefault()
+                        
+                        apiClient({
+                            method: 'get',
+                            url: '/api/admin/word/' + word.id
+                        }).then(response => {
+                            setWord({
+                                ...word,
+                                ...response.data
+                            })
+                        }).catch((error) => {
+                            Toastify("error", error)
+                        })
+                    }}>Edit</button>
+                </td>
+            </tr>
+        })
+    }
+
+    if (isCategoryLoading) {
+        dropdown_options = ""
+    }
+    else {
+        dropdown_options = 
+        <>
+            {
+                categoryList.map((category) => {
+                    return <option key={ category.id } value={ category.id }>{ category.title }</option>
+                })
+            }
+        </>
+    }
```

Comment:
create a function for this one, this is not a good practice. it will rerender if you have any changes on your state. Much better you can memoize this one. Use usecallback or usememo